Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -82,32 +86,69 @@ sub initPlugin {
 
 sub shutdownPlugin {
 	my @played = values %recentlyPlayed;
+
 	$prefs->set('recent', \@played);
 }
 
+sub updateRecentlyPlayed {
+	my ($class, $client, $song) = @_;
+	my ($url) = unwrapUrl($song->originUrl);
+
+	$recentlyPlayed{$url} = { 
+			url      => $url,
+			title    => $song->track->title,
+			cover    => Slim::Player::ProtocolHandlers->iconForURL($url, $client),
+			duration => $song->duration,
+	};	
+}
+
+sub unwrapUrl {
+	return shift =~ m|^podcast://([^{]+)(?:{from=(\d+)}$)?|;	
+}
+
+sub wrapUrl {
+	my ($url, $from) = @_;
+	
+	return 'podcast://' . $url . ($from ? "{from=$from}" : '');
+}
+
 sub handleFeed {
 	my ($client, $cb, $params, $args) = @_;
 
-	# hook in to new song event - show "jump to last position" menu if matching a podcast
-	Slim::Control::Request::subscribe(\&songChangeCallback, [['playlist'], ['newsong', 'pause', 'stop']]);
-
 	my $items = [];
-	
 	my @feeds = @{$prefs->get('feeds')}; 
-	
-	foreach ( @feeds ) {
-		push @$items, {
-			name => $_->{name},
-			url  => $_->{value},
-			parser => 'Slim::Plugin::Podcast::Parser',
-		}
-	}
-	
+
 	push @$items, {
 		name  => cstring($client, 'PLUGIN_PODCAST_RECENTLY_PLAYED'),
 		url   => \&recentHandler,
 		type  => 'link',
+		image => __PACKAGE__->_pluginDataFor('icon'),
 	};
+
+	foreach ( @feeds ) {
+		my $url = $_->{value};
+		my $image = $cache->get('podcast-rss-' . $url);
+
+		push @$items, {
+			name => $_->{name},
+			url  => $url,
+			parser => 'Slim::Plugin::Podcast::Parser',
+			image => $image || __PACKAGE__->_pluginDataFor('icon'),
+		};
+
+		Slim::Networking::SimpleAsyncHTTP->new(
+			sub { 
+				eval {
+					my $xml = XMLin(shift->content);
+					my $image = $xml->{channel}->{image}->{url} || $xml->{channel}->{'itunes:image'}->{href};
+					$cache->set('podcast-rss-' . $url, $image, '90days') if $image;
+				};
```

Comment:
First time we render the menu, there is a default RSS image. I've changed things so that if we fail parsing, we'll store the default image in cache for a day. Other than that, I feel that if there is a general error, we should retry when the user hits the menu. 
I did not wait for all icons to be populated because it only happens once every 90 days, so I'm not sure this is worth more tweaks